packages/dom/src/htmlsink.rs

```diff
@@ -31,20 +30,23 @@ pub struct DocumentHtmlParser<'a> {
 
     /// The document's quirks mode.
     pub quirks_mode: QuirksMode,
+
+    net_provider: N,
```

Comment:
I think we should make this a `Box<dyn NetProvider<Resource>>`, then we could remove the generic parameter which would simplify the code and also ought to help compile times.